README.md

```diff
@@ -497,6 +497,23 @@ Each `extra_arguments` block includes an arbitrary name (in the example above, `
 which the extra arguments should be add, a list of `arguments` or `required_var_files` or `optional_var_files` to add.
 With the configuration above, when you run `terragrunt apply`, Terragrunt will call Terraform as follows:
 
+When available, it is preferable to use interpolation functions such as
+[get_terraform_commands_that_need_locking](#get_terraform_commands_that_need_locking) and
+[get_terraform_commands_that_need_vars](#get_terraform_commands_that_need_vars)
+since they provide the complete list of terraform commands that make use of the desired parameter:
+
+```hcl
+terragrunt = {
+  terraform {
+    # Force Terraform to keep trying to acquire a lock for up to 20 minutes if someone else already has the lock
+    extra_arguments "retry_lock" {
+      commands  = "${get_terraform_commands_that_need_locking()}"
```

Comment:
Would the implementation and docs be simplified if `get_terraform_commands_that_need_locking()` returned a comma-separated list of command names and you the user provided the brackets themselves?

```
commands = ["${get_terraform_commands_that_need_locking()}"]
```